frame/staking/src/pallet/impls.rs

```diff
@@ -1319,3 +1472,187 @@ impl<T: Config> SortedListProvider<T::AccountId> for UseNominatorsMap<T> {
 		}
 	}
 }
+
+/// A static tracker for the snapshot of all voters.
+///
+/// Computes the (SCALE) encoded byte length of a snapshot based on static rules, without any actual
+/// encoding.
+///
+/// ## Warning
+///
+/// Make sure any change to SCALE is reflected here.
+///
+/// ## Details
+///
+/// The snapshot has a the form `Vec<Voter>` where `Voter = (Account, u64, Vec<Account>)`. For each
+/// voter added to the snapshot, [`register_voter`] should be called, with the number of votes
+/// (length of the internal `Vec`).
+///
+/// Whilst doing this, [`size`] will track the entire size of the `Vec<Voter>`, except for the
+/// length prefix of the outer `Vec`. To get the final size at any point, use
+/// [`final_byte_size_of`].
+pub(crate) struct StaticSizeTracker<AccountId> {
```

Comment:
I would make this fully generic and reusable, but it is not really feasible given the nested vec.